### PR TITLE
Use ISO week date numbering for english too

### PR DESCRIPTION
**Pull request by thecoop**
TL;DR - dates are really, really complicated. Reference - https://en.wikipedia.org/wiki/ISO_week_date

With the changes to defaulting the locale to english rather than root (https://github.com/elastic/elasticsearch/pull/112796, https://github.com/elastic/elasticsearch/pull/112799), the Iso override added by https://github.com/elastic/elasticsearch/pull/48209 no longer applies to english, which results in broken tests as the week numbering changes between the two locales. This PR fixes the immediate problem.

However, this opens up a rather large can of worms - what about all the other locales people use? I don't believe we have any defined behaviour around this, so what is the correct thing to do long-term, and do we want to break week numbering too? Do we just revert to whatever CLDR does, with the corresponding break here too?

This is not a straightforward change - there may be users explicitly using `en` locale with non-iso week dates that will be broken by this PR.

### CODE REVIEW

**Comment by thecoop** [comment]
They are both first-week-sunday, 1 day minimum week, on compat and cldr. But we override it for compat root, which is where this discrepancy comes in.

There are also other changes in other locales between compat and cldr.

**Comment by thecoop** [comment]
We're going to remove the Iso implementation instead
